src/cpuid/src/template/intel/mod.rs

```diff
@@ -20,3 +26,29 @@ pub fn validate_vendor_id() -> Result<(), Error> {
 
     Ok(())
 }
+
+/// Returns MSRs to be saved based on the Intel CPUID features that are enabled.
+pub(crate) fn msrs_to_save_by_cpuid(cpuid: &CpuId) -> HashSet<u32> {
```

Comment:
Now I started wondering, is this approach only applicable to a templated execution? I guess we should consult CPUID when choosing which MSRs to save regardless if a template is used or not in general. There will be Intel/AMD specific features/MSRs, but there is a considerable overlap.

Reply:
It's not only applicable to templates and it is used even if we don't use templates. Only the higher level function is exported and at some point we might want to add common things there for X86 general registers.
The reason why I added this functionality in the `template` module is just convenience. Ideally we would want to move all the template stuff in another crate that is separate from CPUID. Not sure if we should move this functionality outside of the template module since it will be used by it too and, in a way, the passthrough CPU configuration is like a template by itself.

But I'm open to suggestions if you think we need to move it somewhere else.

Reply:
Indeed, it is used in non-templated version. I was misled in my analysis. I think the function could stay were it is until we have a better idea of how to structure the MSR/CPUID dependency code.